### PR TITLE
Feature - Summary Bar

### LINKED ISSUE #139: Feature - Summary Bar
To fulfil the users need to manage applications at a higher level, we have decided to implement a summary bar on the side for the user to have relevant statistics to keep track of the overall applications.

Example summary statistics to keep track of:
- Total number of applications
- Number of `HIGH`, `MEDIUM` and `LOW` priority applications
- Number of `APPLIED` and `NOT_APPLIED` applications

### PR DESCRIPTION
**Pull request by joszx**
# Added Summary Bar feature

## Key Summary
- Summary Bar is now operational
- Keeps track of total number of applications, applications with `HIGH`, `MEDIUM`, `LOW` priority tag and applications with  `APPLIED`, `NOT_APPLIED` application status tags
- Initializes on application startup and updates on every command

## How it functions
`SummaryList` stores all `SummaryBox` and feeds each of them the relevant statistic. When a command is executed, provided the command is a valid command, `SummaryList` will be updated by passing in the new list of applications for it to get the relevant statistics and update with a new list of `SummaryBox`.

## Preview
image

## Other Updates
- Fix README.md and index.md UI image link to the updated MainWindowUI.png
- Manually tested each command with respect to the summary bar
- Automated tests passed 267/267

## Further improvements
1. Update UG/DG with summary bar feature
2. Add automated test cases for summary bar model and UI components
3. Improve on the summary bar GUI e.g. changing fonts, text color, box outline etc

Ready for review @lchokhoe @tanyutao544

### CODE REVIEW

**Comment by joszx** [comment]
Fix #139

File: src/main/java/seedu/address/model/summarybar/SummaryList.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package seedu.address.model.summarybar;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.application.Application;
+import seedu.address.model.tag.Tag;
+
+/**
+ * A list of SummaryBoxes that keeps track of their respective indices and tags.
+ *
```
Comment (lchokhoe) [inline src/main/java/seedu/address/model/summarybar/SummaryList.java:10]: Extra line here can be removed.

File: src/main/java/seedu/address/model/summarybar/TagSummaryBox.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        super(name, totalApplications);
+        this.currApplications = currApplications;
+    }
+
+    @Override
+    public Optional<Integer> getCurrApplications() {
+        return Optional.of(currApplications);
+    }
+
+
```
Comment (lchokhoe) [inline src/main/java/seedu/address/model/summarybar/TagSummaryBox.java:29]: Extra line can be removed.

File: src/main/java/seedu/address/model/summarybar/TotalApplicationsBox.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package seedu.address.model.summarybar;
+
+import java.util.Optional;
+
+/**
+ * Represents a SummaryBox that tracks the total number of applications in InternApply.
+ */
+public class TotalApplicationsBox extends SummaryBox {
```
Comment (lchokhoe) [inline src/main/java/seedu/address/model/summarybar/TotalApplicationsBox.java:8]: Naming of this class could follow TagSummaryBox.java

Maybe TotalApplicationsSummaryBox.java?

File: src/main/resources/view/SummaryListPanel.fxml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<?import javafx.scene.control.ListView?>
+<?import javafx.scene.layout.VBox?>
+
+<VBox xmlns="http://javafx.com/javafx/8" xmlns:fx="http://javafx.com/fxml/1">
```
Comment (lchokhoe) [inline src/main/resources/view/SummaryListPanel.fxml:6]: Not sure if you missed this line but xmlns="http://javafx.com/javafx8" is xmlns="http://javafx.com/javafx11" in the other .fxml files

File: README.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 [![Java CI](https://github.com/AY2122S2-CS2103T-T11-3/tp/actions/workflows/gradle.yml/badge.svg)](https://github.com/AY2122S2-CS2103T-T11-3/tp/actions/workflows/gradle.yml)
 [![codecov](https://codecov.io/gh/AY2122S2-CS2103T-T11-3/tp/branch/master/graph/badge.svg?token=OPX1FSESUJ)](https://codecov.io/gh/AY2122S2-CS2103T-T11-3/tp)
 
-![Ui](docs/images/Ui.png)
+![Ui](docs/images/MainWindowUi.png)
```
Comment (lchokhoe) [inline README.md:4]: Good catch!

File: docs/index.md [hunk 6]
```diff
@@ -5,7 +5,7 @@ title: InternApply
 
 [![CI Status](https://github.com/se-edu/addressbook-level3/workflows/Java%20CI/badge.svg)](https://github.com/AY2122S2-CS2103T-T11-3/tp/actions)
 [![codecov](https://codecov.io/gh/AY2122S2-CS2103T-T11-3/tp/branch/master/graph/badge.svg?token=OPX1FSESUJ)](https://codecov.io/gh/AY2122S2-CS2103T-T11-3/tp)
-![Ui](images/Ui.png)
+![Ui](images/MainWindowUi.png)
```
Comment (lchokhoe) [inline docs/index.md:8]: Good catch!

**Review by lchokhoe — COMMENTED** [review]
I left some comments however they are mostly minor stuff.

The rest of the code LGTM!

**Review by lchokhoe — APPROVED** [review]
LGTM.

**Pull request merged**